include/cmr_geometry_utils/transforms_utils.hpp

```diff
@@ -13,7 +13,7 @@
 
 namespace cmr_geometry_utils {
 
-inline bool transform_pose(const std::shared_ptr<tf2_ros::Buffer> tf, const std::string frame,
+inline bool transform_pose(const std::shared_ptr<tf2_ros::Buffer> & tf, const std::string & frame,
```

Comment:
I don't think the reference to the pointer makes sense

Reply:
"smart pointers" are not pointers, but objects.  The general rule is that if the receiver is not participating in the lifecycle of the "pointer" then it should be passed by const ref.
Even if it was participating, this would typically be done by another copy.

Performance wise, copying smart pointers is not free.  Each copy is an atomic increment of an internal reference counter, so that the item can be used simultaneously in multiple threads without extra bookkeeping burden placed upon the developer.  While the cost is small, its not nothing, and its noticeable when called in a tight loop.

Typically it is values that are not passed by ref, numbers, small structs, enums, pointers etc

Reply:
Interesting point [on passing a smart pointer](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-sharedptrparam-const), I've never thought about it from that perspective.

Reply:
Does passing a ref to the smart pointer increase the internal counter?

Reply:
it does not, its similar to passing a pointer, the object is unaware of who is pointing at it.  A ref is 100% identical to the original object, no mutations occur

Reply:
As we already started that topic :)

shared_ptr size is 16 bytes on modern architectures ([link](https://godbolt.org/z/s1qnxcEfT)), so it basically size of 2 doubles. Looks like it won't create much performance overhead while copying (we should add a ref counter increment of course). What do you think, @cmraaron ? Is that valid reasoning?

Reply:
its not about the size of the object.  its about the semantics.  the explanation re ref counting was just background, not justification for this particular change.  performance is a non issue here.  In the C++ guidelines linked above, its explained, rather cryptically, what the various signatures communicate to the reader.  Its explained with a bit more detail here https://www.modernescpp.com/index.php/c-core-guidelines-passing-smart-pointer